test/externalTests/plugins/testCommon.js

```diff
@@ -30,22 +31,24 @@ function inject (bot) {
   }
 
   const superflatLayers = [
-    new Block(mcData.blocksByName.bedrock.id),
-    new Block(mcData.blocksByName.dirt.id),
-    new Block(mcData.blocksByName.dirt.id),
-    new Block(mcData.blocksByName.grass.id)
+    { block: new Block(mcData.blocksByName.bedrock.id), item: new Item(mcData.itemsByName.bedrock.id) },
```

Comment:
Why test fails

Reply:
For versions prior to 1.13, minecraft-data items.json doesn't contain the blocks, only the items.
https://github.com/PrismarineJS/minecraft-data/blob/master/data/pc/1.12/items.json
https://github.com/PrismarineJS/minecraft-data/blob/master/data/pc/1.13.2/items.json
Changing the lines to:
```javascript
  const superflatLayers = [
    { block: new Block(mcData.blocksByName.bedrock.id), item: new Item(mcData.blocksByName.bedrock.id) },
    { block: new Block(mcData.blocksByName.dirt.id), item: new Item(mcData.blocksByName.dirt.id) },
    { block: new Block(mcData.blocksByName.dirt.id), item: new Item(mcData.blocksByName.dirt.id) },
    { block: new Block(mcData.blocksByName.grass.id), item: new Item(mcData.blocksByName.grass.id) }
    // and then air
  ]
```
I'm able to run the tests.

Reply:
Apparently, in 1.13, certain blocks don't have the same id as their item (for instance grass block have the id of brown wool item)
```
[
  {
    block: Block {
      type: 25,
      metadata: undefined,
      light: 0,
      skyLight: 0,
      biome: [Biome],
      position: null,
      stateId: 33,
      name: 'bedrock',
      hardness: null,
      displayName: 'Bedrock',
      boundingBox: 'block',
      diggable: false,
      material: undefined,
      harvestTools: undefined,
      drops: [Array]
    },
    item: Item {
      type: 25,
      count: undefined,
      metadata: 0,
      nbt: null,
      name: 'bedrock',
      displayName: 'Bedrock',
      stackSize: 64
    }
  },
  {
    block: Block {
      type: 9,
      metadata: undefined,
      light: 0,
      skyLight: 0,
      biome: [Biome],
      position: null,
      stateId: 10,
      name: 'dirt',
      hardness: 0.5,
      displayName: 'Dirt',
      boundingBox: 'block',
      diggable: true,
      material: 'dirt',
      harvestTools: undefined,
      drops: [Array]
    },
    item: Item {
      type: 9,
      count: undefined,
      metadata: 0,
      nbt: null,
      name: 'dirt',
      displayName: 'Dirt',
      stackSize: 64
    }
  },
  {
    block: Block {
      type: 9,
      metadata: undefined,
      light: 0,
      skyLight: 0,
      biome: [Biome],
      position: null,
      stateId: 10,
      name: 'dirt',
      hardness: 0.5,
      displayName: 'Dirt',
      boundingBox: 'block',
      diggable: true,
      material: 'dirt',
      harvestTools: undefined,
      drops: [Array]
    },
    item: Item {
      type: 9,
      count: undefined,
      metadata: 0,
      nbt: null,
      name: 'dirt',
      displayName: 'Dirt',
      stackSize: 64
    }
  },
  {
    block: Block {
      type: 94,
      metadata: undefined,
      light: 0,
      skyLight: 0,
      biome: [Biome],
      position: null,
      stateId: 1041,
      name: 'grass',
      hardness: 0,
      displayName: 'Grass',
      boundingBox: 'block',
      diggable: true,
      material: 'plant',
      harvestTools: undefined,
      drops: [Array]
    },
    item: Item {
      type: 94,
      count: undefined,
      metadata: 0,
      nbt: null,
      name: 'brown_wool',
      displayName: 'Brown Wool',
      stackSize: 64
    }
  }
]
```
So there is a reason for the way it is, however it breaks the tests for older versions. I see 2 solutions:
* Patch minecraft-data for versions prior to 1.13 so that the blocks are also in itemsByName
* Add version specific code in mineflayer

Reply:
In fact, the grass block was renamed in 1.13 (it is now named grass_block instead of grass).
```
{
    block: Block {
      type: 8,
      metadata: undefined,
      light: 0,
      skyLight: 0,
      biome: [Biome],
      position: null,
      stateId: 8,
      name: 'grass_block',
      hardness: 0.6,
      displayName: 'Grass Block',
      boundingBox: 'block',
      diggable: true,
      material: 'dirt',
      harvestTools: undefined,
      drops: [Array]
    },
    item: Item {
      type: 8,
      count: undefined,
      metadata: 0,
      nbt: null,
      name: 'grass_block',
      displayName: 'Grass Block',
      stackSize: 64
    }
  }
```
With the correct block, the block and item id are the same. But changing the name from grass to grass_block breaks backward compatibility in another way

Reply:
Are you sure this is the right way to solve this ? Could you check the way blocks and items are defined from 1.13 in wiki.vg (it's called "the flattening") and check if it fits the current code ?

Reply:
Thank you for working on this !

Reply:
In 1.13, not all blocks have their id equal to their item id (eg: brown_wool block=113 item=94) but it is the case for bedrock, dirt and grass_block.

I think the best way would be to patch minecraft-data (or node-minecraft-data) to populate itemsByName with the missing blocks for versions prior to 1.13. What do you think?

Reply:
No I think this is a hack.
There are 3 cases :
1. You want to retrieve a block by id
2. You want to retrieve an item by id
3. You want to retrieve a corresponding block given an item I'd

We need functions to retrieve the 3.
I believe we already have 1 and 2.
To do 3, I think the best way is simply to have a block_item_mapping.json file in minecraft data. This file can be built very easily and with a little bit more effort for 1.13

What do you think ?

Reply:
And for the name, that should be handled in mineflayer directly